pyproject.toml

```diff
@@ -3,39 +3,49 @@ requires = ["setuptools>=61.0"]
 build-backend = "setuptools.build_meta"
 
 [project]
-name = "ospsd-hw1-python"
+name = "HW1_OSPSD"
 version = "0.1.0"
-description = "OSPSD HW1 Python Project"
+description = "hw1 Calculator"
 readme = "README.md"
 requires-python = ">=3.9"
-dependencies = [
+
+[project.optional-dependencies]
+dev = [
     "pytest>=7.0.0",
+    "pytest-mock>=3.0.0",
+    "mypy>=1.0.0",
     "pytest-cov>=4.1.0",
-    "pytest-html>=4.0.0"
+    "pytest-html>=4.0.0",
+    "ruff>=0.3.0",
+    "types-setuptools>=42.0.0",
+]
+
+[tool.ruff.lint]
+select = ["ALL"]
+ignore = [
+    "E501",  # Ignore line length issues since no need to go over 100 characters
+    "W293", # White spaces are used for indentation in doc strings
```

Comment:
i don't understand these two... can you provide an example of what fails without them?